iceoryx_hoofs/test/moduletests/test_cxx_function.cpp

```diff
@@ -252,9 +253,16 @@ TEST_F(function_test, ConstructionFromAnotherFunctionIsCallable)
 TEST_F(function_test, FunctionStateIsIndependentOfSource)
 {
     ::testing::Test::RecordProperty("TEST_ID", "8302046f-cd6a-4527-aca6-3e6408f87a6b");
-    constexpr uint32_t INITIAL_STATE = 73U;
-    static_storage<1024U> storage;
-    auto* p = storage.allocate<Functor>();
+    constexpr uint32_t INITIAL_STATE = 73;
+    constexpr uint32_t MEMORY_SIZE = 1024;
+
+    // NOLINTNEXTLINE(hicpp-no-malloc, cppcoreguidelines-no-malloc) low-level memory management for testing purpose
```

Comment:
@elBoberido 

I was not aware that we enforce static analysis rules in test code as well. I think we should not (we already established that we use certain constructs in tests that are not allowed in library code, such as dynamic memory). If we do enforce the rules we need the comment of course.

Reply:
@MatthiasKillat we enforce a subset of what we do for production code. In the past we also had some issues in the tests which at some point resulted in weird test failures. If the current list of checks is too strict for the tests, we can of course talk about relaxing them.

Reply:
I see. Dynamic memory allocation might be one of them, since we allow `new` and `delete` for objects created in tests `malloc` and `free` should be allowed for similar reasons without a comment.